exceptions/include/peach_library/exceptions/utils/exception_format.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef PEACH_EXCEPTION_FORMAT_HPP
+#define PEACH_EXCEPTION_FORMAT_HPP
+
+#include <ctime>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "exception_safety.hpp"
+
+namespace peach::detail
+{
+  template< OutputStreamInsertable... Tys >
+  std::string format_error( const int line_num, const std::string& file_name,
+                            const std::string& prefix, Tys&&... args )
+  {
+    std::stringstream ss { };
+
+    const time_t t { std::time( nullptr ) };
+    tm current_time { };
+    localtime_s( &current_time, &t );
```

Comment:
IIRC `localtime_s` does not exist on Linux. So when ensuring Linux compatibility we will need to check for the platform we compile for. Therefore i would vote to create a function that returns the time in which we can do the platform dependent stuff .